src/python/pants/rules/core/list_target_types.py

```diff
@@ -32,56 +34,87 @@ class TargetTypes(Goal):
 
-    fields = target_type.class_field_types(union_membership=union_membership)
-    longest_field_name = max(len(field.alias) for field in fields)
+@dataclass(frozen=True)
+class FieldInfo:
+    alias: str
+    description: Optional[str]
+    type_hint: str
+    default: str
 
-    # TODO: consider hard wrapping fields. It's confusing to read when fields soft wrap.
-    def format_field(field: Type[Field]) -> str:
-        field_alias_text = f"  {field.alias}"
-        field_alias = console.cyan(f"{field_alias_text:<{longest_field_name + 2}}")
-        description = get_docstring(field, flatten=True) or ""
+    @classmethod
+    def create(cls, field: Type[Field]) -> "FieldInfo":
+        description = get_docstring(field, flatten=True)
         if issubclass(field, PrimitiveField):
             raw_value_type = get_type_hints(field.compute_value)["raw_value"]
         elif issubclass(field, AsyncField):
             raw_value_type = get_type_hints(field.sanitize_raw_value)["raw_value"]
         else:
             raw_value_type = get_type_hints(field.__init__)["raw_value"]
-        type_info = console.green(
-            f"(type: {pretty_print_type_hint(raw_value_type)}, default: {field.default})"
+        type_hint = pretty_print_type_hint(raw_value_type)
+        return cls(
+            alias=field.alias,
+            description=description,
+            type_hint=type_hint,
+            default=str(field.default),
+        )
+
+    def format_for_cli(self, *, console: Console) -> str:
+        field_alias = console.magenta(f"{self.alias}")
+        type_info = console.cyan(f"  type: {self.type_hint}, default: {self.default}")
+        lines = [field_alias, type_info]
+        if self.description:
+            lines.extend(f"  {line}" for line in textwrap.wrap(self.description, 80))
+        return "\n".join(f"  {line}" for line in lines)
+
+
+@dataclass(frozen=True)
+class VerboseTargetInfo:
+    alias: str
+    description: Optional[str]
+    fields: Sequence[FieldInfo]
+
+    @classmethod
+    def create(
+        cls, target_type: Type[Target], *, union_membership: UnionMembership
+    ) -> "VerboseTargetInfo":
+        return cls(
+            alias=target_type.alias,
+            description=get_docstring(target_type),
+            fields=[
+                FieldInfo.create(field)
+                for field in target_type.class_field_types(union_membership=union_membership)
+            ],
         )
-        type_info_prefix = " " if description else ""
-        return f"{field_alias}  {description}{type_info_prefix}{type_info}"
-
-    output = [target_type_description, "\n"] if target_type_description else []
-    output.extend(
-        [
-            console.blue(f"{target_type.alias}("),
-            *sorted(format_field(field) for field in fields),
-            console.blue(")"),
+
+    def format_for_cli(self, *, console: Console) -> str:
+        output = [
+            console.green(f"{self.alias}()"),
```

Comment:
I'm in no way wedded to using the parentheses, e.g. `python_library()`. It looked better than `python_library` and `python_library:` imo, but open to suggestions.

Reply:
However it settles, more important than the puntuation or lack thereof, a blank line separating the target name from the description seems like it would go a long way.